pkg/destroy/aws/aws.go

```diff
@@ -2026,3 +2029,121 @@ func isBucketNotFound(err interface{}) bool {
 	}
 	return false
 }
+
+func deleteElasticFileSystem(ctx context.Context, session *session.Session, arn arn.ARN, logger logrus.FieldLogger) error {
+	client := efs.New(session)
+
+	resourceType, id, err := splitSlash("resource", arn.Resource)
+	if err != nil {
+		return err
+	}
+
+	switch resourceType {
+	case "file-system":
+		return deleteFileSystem(ctx, client, id, logger)
+	default:
+		return errors.Errorf("unrecognized elastic file system resource type %s", resourceType)
+	}
+}
+
+func deleteFileSystem(ctx context.Context, client *efs.EFS, fsid string, logger logrus.FieldLogger) error {
+	logger = logger.WithField("Elastic FileSystem ID", fsid)
+
+	// Delete all MountTargets + AccessPoints under given FS ID
+	mountTargetIDs, err := getMountTargets(ctx, client, fsid)
+	if err != nil {
+		return err
+	}
+	for _, mt := range mountTargetIDs {
+		err := deleteMountTarget(ctx, client, mt, logger)
+		if err != nil {
+			return err
+		}
+	}
+	accessPointIDs, err := getAccessPoints(ctx, client, fsid)
+	if err != nil {
+		return err
+	}
+	for _, ap := range accessPointIDs {
+		err := deleteAccessPoint(ctx, client, ap, logger)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = client.DeleteFileSystemWithContext(ctx, &efs.DeleteFileSystemInput{FileSystemId: aws.String(fsid)})
+	if err != nil {
+		return err
```

Comment:
Is there a case here where the error is that the file system doesn't exist (as it may have been previously deleted)? If you look elsewhere in the code, we handle this differently from other errors.

Reply:
Fixed everywhere